src/offline/wallet.ts

```diff
@@ -0,0 +1,88 @@
+import {fromSeed} from 'bip32'
+import {mnemonicToSeedSync} from 'bip39'
+import {decode, encodeForSigning} from 'ripple-binary-codec'
+import {deriveKeypair, generateSeed, verify} from 'ripple-keypairs'
+import {signOffline} from '../transaction/sign'
+import {SignOptions} from '../transaction/types'
+
+export interface WalletOptions {
+  // The entropy to use to generate the seed.
+  entropy?: Uint8Array | number[]
+
+  // The digital signature algorithm to generate an address for. Can be `ecdsa-secp256k1` (default) or `ed25519`.
+  algorithm?: 'ecdsa-secp256k1' | 'ed25519'
```

Comment:
i think this should be a type defined someone outside this class. why call it `ecdsa-secp256k1` instead of `secp256k1`?

Reply:
also - this is misleading since only some functions use `entropy` and only others use `derivationPath`. Sharing them as a single options type isn't helping, but actually making it less clear, to me

Reply:
> i think this should be a type defined someone outside this class. why call it `ecdsa-secp256k1` instead of `secp256k1`?

Yes, initially I was going to include the refactor of replacing string literals `ecdsa-secp256k1` and `ed25519` to be referenced as enum types throughout our codebase. However, these file changes seemed like it would make this PR hard to read. If you want I can include these changes here or we can do it in a separate PR to define and apply ECDSA enum type??

Reply:
> also - this is misleading since only some functions use `entropy` and only others use `derivationPath`. Sharing them as a single options type isn't helping, but actually making it less clear, to me

Ah I see, one way to refactor this is to remove this single `WalletOptions` param and instead have `Wallet` functions take in appropriate params. I'll go with this approach if that sounds good to you. 🙂

Reply:
> i think this should be a type defined someone outside this class. why call it `ecdsa-secp256k1` instead of `secp256k1`?

I decided to create a new enum called ECDSA that contains these algorithms as its values. It's in this updated PR now.